File: 프로그래머스/1/92334. 신고 결과 받기/신고 결과 받기.py
def solution(id_list, report, k):    
    report_uniq = dict.fromkeys(report)
    report_count = dict.fromkeys(id_list, 0)
    id_count = dict.fromkeys(id_list, 0)
    
    for rp in report_uniq:
        a, b = rp.split()
        report_count[b] += 1     
    report_count = {key: value for key, value in report_count.items() if value >= k } 
    
    for rp in report_uniq:
        a, b = rp.split()
        for rc in report_count:
            if rc == b:
                id_count[a] += 1
                
    return list(id_count.values())

File: 프로그래머스/1/92334. 신고 결과 받기/신고 결과 받기.cpp
#include <string>
#include <vector>
#include <sstream>
#include <unordered_map>
#include <unordered_set>

using namespace std;

vector<int> solution(vector<string> id_list, vector<string> report, int k) {
    vector<int> answer;
    unordered_map<string, int> report_dict;
    unordered_set<string> uniq_report;
    unordered_map<string, vector<string>> report_user;
    
    for(string rp : report) uniq_report.insert(rp);
    
    for (string rp : uniq_report) {
        stringstream ss(rp);
        string id, rpid;
        ss >> id >> rpid;
        report_dict[rpid] += 1;
        report_user[id].push_back(rpid);
    }
    
    for (string id : id_list){
        answer.push_back(0);
        for (string rpid : report_user[id]){
            if (report_dict[rpid] >= k)
                answer.back()++;
        }
    }

    return answer;
}